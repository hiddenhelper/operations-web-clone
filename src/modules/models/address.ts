import { INamedEntity } from './general';

export enum AddressFields {
  LINE_1 = 'line1',
  LINE_2 = 'line2',
  COUNTRY = 'country',
  COUNTRY_ID = 'countryId',
  CITY = 'city',
  STATE = 'stateCode',
  STATE_NAME = 'stateName',
  ZIP_CODE = 'zipCode',
  COUNTY = 'county',
  BOROUGH = 'borough',
  LATITUDE = 'latitude',
  LONGITUDE = 'longitude',
  ATTENTION_1 = 'attention1',
  ATTENTION_2 = 'attention2',
}

export enum LocationsFields {
  NAME = 'name',
}

export enum Borough {
  MANHATTAN = 'Manhattan',
  BROOKLYN = 'Brooklyn',
  QUEENS = 'Queens',
  STATEN_ISLAND = 'Staten Island',
  THE_BRONX = 'The Bronx',
}

export interface IAddress {
  [AddressFields.LINE_1]: string;
  [AddressFields.LINE_2]: string;
  [AddressFields.CITY]: string;
  [AddressFields.STATE]: string;
  [AddressFields.ZIP_CODE]: string;
  [AddressFields.COUNTRY]?: INamedEntity;
  [AddressFields.COUNTRY_ID]?: string;
  [AddressFields.COUNTY]?: string;
  [AddressFields.ATTENTION_1]?: string;
  [AddressFields.ATTENTION_2]?: string;
  [AddressFields.STATE_NAME]?: string;
  [AddressFields.BOROUGH]?: Borough;
  [AddressFields.LATITUDE]: number;
  [AddressFields.LONGITUDE]: number;
}

export interface ILocation {
  id?: string;
  [LocationsFields.NAME]?: string;
  isUsed?: boolean;
}

export const addressFieldRules = {
  required: [
    { name: AddressFields.LINE_1, required: true },
    { name: AddressFields.LINE_2, required: false },
    { name: AddressFields.COUNTRY_ID, required: true, computePositive: true },
    { name: AddressFields.STATE, required: true },
    { name: AddressFields.ZIP_CODE, required: true },
    { name: AddressFields.CITY, required: true },
    { name: AddressFields.ATTENTION_1, required: false },
    { name: AddressFields.ATTENTION_2, required: false },
  ],
  notRequired: [
    { name: AddressFields.LINE_1, required: false },
    { name: AddressFields.LINE_2, required: false },
    { name: AddressFields.COUNTRY_ID, required: false },
    { name: AddressFields.STATE, required: false },
    { name: AddressFields.ZIP_CODE, required: false },
    { name: AddressFields.CITY, required: false },
    { name: AddressFields.ATTENTION_1, required: false },
    { name: AddressFields.ATTENTION_2, required: false },
  ],
};

export const boroughOptionList = [
  {
    value: Borough.MANHATTAN,
    label: Borough.MANHATTAN,
  },
  {
    value: Borough.BROOKLYN,
    label: Borough.BROOKLYN,
  },
  {
    value: Borough.QUEENS,
    label: Borough.QUEENS,
  },
  {
    value: Borough.STATEN_ISLAND,
    label: Borough.STATEN_ISLAND,
  },
  {
    value: Borough.THE_BRONX,
    label: Borough.THE_BRONX,
  },
];

export enum ZipCodeFormatType {
  FIVE_DIGITS = 0,
  FIVE_DIGITS_PLUS_FOUR = 1,
  CANADA_FORMAT = 2,
}

export const zipCodeFormatMap: { [key: string]: { placeholder: string; mask: any[] } } = {
  [ZipCodeFormatType.FIVE_DIGITS]: {
    placeholder: '00000',
    mask: [/\d/, /\d/, /\d/, /\d/, /\d/, /\d/],
  },
  [ZipCodeFormatType.FIVE_DIGITS_PLUS_FOUR]: {
    placeholder: '00000-0000',
    mask: [/\d/, /\d/, /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  },
  [ZipCodeFormatType.CANADA_FORMAT]: {
    placeholder: 'ANA-NAN',
    mask: [/[a-zA-Z]/, /\d/, /[a-zA-Z]/, '-', /\d/, /[a-zA-Z]/, /\d/],
  },
};

export const getFallbackAddress = () => ({
  line1: null,
  line2: null,
  city: null,
  countryId: null,
  stateCode: null,
  zipCode: null,
  county: null,
  latitude: null,
  longitude: null,
  stateName: null,
});

export const getFallbackLocation = (): ILocation => ({
  name: null,
});
