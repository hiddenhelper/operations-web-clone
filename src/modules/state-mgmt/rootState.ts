import { History } from 'history';

import { IState as IAuthState } from './auth/state';
import { IState as IGeneralState } from './general/state';
import { IState as IClientState } from './client/state';
import { IState as IProjectState } from './project/state';
import { IState as IUserState } from './user/state';
import { IState as IAccessControlSystemState } from './access-control-system/state';
import { IState as IBadgePrintingSystemState } from './badge-printer-system/state';
import { IState as IWorkerState } from './worker/state';
import { IState as ICertificationState } from './certification/state';
import { IState as IFileState } from './file/state';
import { IState as IBadgeState } from './badge/state';
import { IState as IStatisticsState } from './statistics/state';
import { IState as ITrainingState } from './training/state';
import { IState as IPaymentState } from './payment/state';
import { IState as IInvoiceState } from './invoice/state';
import { ApiService } from '../services/ApiService';
import { AuthService } from '../services/AuthService';
import { PrinterService } from '../services/PrinterService';
import { FileService } from '../services/FileService';
import { Logger } from '../services/Logger';
import { IProcoreState } from './procore/state';
import { IState as IProjectNewState } from './project-new/state';

export interface IAction {
  type: string;
  payload: any;
}

export interface IRootState {
  auth: IAuthState;
  user: IUserState;
  client: IClientState;
  project: IProjectState;
  accessControlSystem: IAccessControlSystemState;
  badgePrinterSystem: IBadgePrintingSystemState;
  worker: IWorkerState;
  certification: ICertificationState;
  training: ITrainingState;
  invoice: IInvoiceState;
  file: IFileState;
  badge: IBadgeState;
  statistics: IStatisticsState;
  payment: IPaymentState;
  general: IGeneralState;
  router?: { location: { pathname: string; search: string; hard: string; key: string }; action: string };
  procore: IProcoreState;
  projectNew: IProjectNewState;
}

export interface IEpicDependencies {
  history: History;
  apiService: ApiService;
  authService: AuthService;
  printerService: PrinterService;
  fileService: FileService;
  logger: typeof Logger;
}
