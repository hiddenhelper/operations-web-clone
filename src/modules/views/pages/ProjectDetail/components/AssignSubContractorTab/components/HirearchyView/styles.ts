import { createStyles, makeStyles } from '@material-ui/core/styles';
import { STYLE } from '../../../../../../../../constants';

export const useStyles = makeStyles(theme =>
  createStyles({
    container: {
      display: 'flex',
      flexDirection: 'column',
    },
    nodeContainer: {
      display: 'flex',
      flexDirection: 'column',
      width: '100%',
    },
    subLevelContainer: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'flex-end',
      '& .nodeContainer': {
        paddingLeft: '5%',
      },
    },
    node: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: STYLE.COLOR.WHITE,
      padding: '12px 18px',
      boxSizing: 'border-box',
      height: '60px',
      border: '1px solid #E5E5E5',
      borderRadius: '5px',
      marginBottom: '10px',
      justifyContent: 'space-between',
    },
    nodeContent: {
      display: 'flex',
      alignItems: 'center',
      '& > div': {
        marginRight: '10px',
      },
      '& .MuiIconButton-root:focus': {
        outline: '0 !important',
      },
      maxWidth: '90%',
    },
    nodeStatus: {
      display: 'flex',
    },
    nodeTrades: {
      fontWeight: 500,
      color: '#444444',
      fontSize: '15px',
      whiteSpace: 'nowrap',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      maxWidth: '65%',
    },
    role: {
      marginLeft: '-20px',
    },
  })
);
