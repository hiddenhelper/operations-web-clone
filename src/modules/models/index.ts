import * as GeneralModel from './general';
import * as UserModel from './user';
import * as ClientModel from './client';
import * as ProjectModel from './project';
import * as ResourceModel from './resource';
import * as WorkerModel from './worker';
import * as AccessControlSystemModel from './accessControlSystem';
import * as BadgePrintingSystemModel from './badgePrinterSystem';
import * as DeviceModel from './device';
import * as InventoryModel from './inventory';
import * as BadgeModel from './badge';
import * as CertificationModel from './certification';
import * as FileModel from './file';
import * as AddressModel from './address';
import * as ConsentFormModel from './consentForm';
import * as StatisticsModel from './statistics';
import * as TrainingModel from './training';
import * as PaymentModel from './payment';
import * as InvoiceModel from './invoice';
import * as ProcoreModel from './procore';
import * as SearchModel from './search';
import * as ProjectNewModel from './project-new';

export {
  GeneralModel,
  UserModel,
  ClientModel,
  ProjectModel,
  ResourceModel,
  AccessControlSystemModel,
  WorkerModel,
  BadgePrintingSystemModel,
  DeviceModel,
  InventoryModel,
  BadgeModel,
  CertificationModel,
  FileModel,
  AddressModel,
  ConsentFormModel,
  StatisticsModel,
  TrainingModel,
  PaymentModel,
  InvoiceModel,
  ProcoreModel,
  SearchModel,
  ProjectNewModel,
};
