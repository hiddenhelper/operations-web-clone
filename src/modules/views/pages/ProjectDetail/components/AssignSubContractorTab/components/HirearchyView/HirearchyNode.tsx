import React, { memo, useMemo, useCallback } from 'react';
import { IconButton } from '@material-ui/core';
import { ChevronRight, ExpandMore } from '@material-ui/icons';

import RoleGuard from 'modules/views/shared/RoleGuard';
import RolePill from 'modules/views/shared/RolePill';
import StatusChip from 'modules/views/shared/StatusChip';
import TableCellLink from 'modules/views/shared/TableCellLink';

import { ClientModel, ProjectModel, UserModel } from 'modules/models';
import { listGlobalStyles } from 'assets/styles';
import { useStyles as statusChipStyles } from '../../../../../../shared/StatusChip/styles';
import { useStyles } from './styles';

export interface IHirearchyNodeProps {
  client: ClientModel.IHirearchyClientProject;
  isSpread: boolean;
  showChilds: (id: string) => void;
}

const HirearchyNode = ({ client, isSpread, showChilds }: IHirearchyNodeProps) => {
  const classes = useStyles();
  const statusChipClasses = statusChipStyles();
  const listClasses = listGlobalStyles();

  const clientProjectStatusMap = useMemo(
    () => ({
      [ProjectModel.CompanyProjectStatus.PENDING]: { label: 'Pending', class: statusChipClasses.pending },
      [ProjectModel.CompanyProjectStatus.ACCEPTED]: { label: 'Accepted', class: statusChipClasses.active },
      [ProjectModel.CompanyProjectStatus.REJECTED]: { label: 'Rejected', class: statusChipClasses.expired },
    }),
    [statusChipClasses]
  );

  const onClick = useCallback(() => {
    showChilds(client.id);
  }, [client.id, showChilds]);

  return (
    <div className={classes.node}>
      <div className={classes.nodeContent}>
        <div className="spread-icon" style={{ minWidth: '30px' }}>
          {!!client.sponsoredCompaniesIds.length && (
            <IconButton data-testid="open-hirearchy-child-list" onClick={onClick} disableRipple={true} aria-label="hirearchy-client-node" color="default">
              {isSpread ? <ExpandMore /> : <ChevronRight />}
            </IconButton>
          )}
        </div>
        <div className={classes.role}>
          <RolePill role={client.role} />
        </div>
        <div className={`${listClasses.listName} ${listClasses.listGeneralText}`}>
          <RoleGuard roleList={[UserModel.Role.FCA_ADMIN]} fallback={<>{client.name}</>}>
            <TableCellLink href={`/clients/detail/${client.id}`} testId="client-navigate-btn" text={client.name} title="View Client details" />
          </RoleGuard>
        </div>
        <div className="role">{ProjectModel.roleMap[client.role]}</div>
        <div className={classes.nodeTrades}>{client.trades.map(trade => trade.name).join(', ')}</div>
      </div>
      <div className={classes.nodeStatus} style={{ display: 'flex' }}>
        <StatusChip
          styleClasses={`${clientProjectStatusMap[client.companyProjectStatus].class} `}
          label={clientProjectStatusMap[client.companyProjectStatus].label}
        />
      </div>
    </div>
  );
};

export default memo(HirearchyNode);
