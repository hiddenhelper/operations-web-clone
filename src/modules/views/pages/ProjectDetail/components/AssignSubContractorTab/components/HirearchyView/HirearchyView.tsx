import React, { memo, useMemo, useState, useCallback, useEffect } from 'react';

import HirearchyNode from './HirearchyNode';

import { GeneralModel, ClientModel } from '../../../../../../../models';
import { toggleArray } from '../../../../../../../../utils/generalUtils';
import { useStyles } from './styles';

export interface IHirearchyViewProps {
  projectId: string;
  hirearchyMap: GeneralModel.IEntityMap<ClientModel.IHirearchyClientProject>;
  loading: GeneralModel.ILoadingStatus;
  assignClientLoading: GeneralModel.ILoadingStatus;
  fetchProjectClientHirearchyList: (id: string) => void;
  closeModal: () => void;
}

const HirearchyView = ({ projectId, hirearchyMap, loading, assignClientLoading, closeModal, fetchProjectClientHirearchyList }: IHirearchyViewProps) => {
  const classes = useStyles();

  const hirearchyList = useMemo(() => Object.values(hirearchyMap).filter(client => client.level === 0), [hirearchyMap]);

  const [childrenList, setChildrenList] = useState<string[]>([]);

  const showChildrenNode = useCallback(
    (id: string) => {
      setChildrenList(prevList => toggleArray(prevList, id));
    },
    [setChildrenList]
  );

  const drawNode = useCallback(
    node => {
      const isSpread = childrenList.includes(node.id);
      return (
        <div key={node.id} className={`${classes.nodeContainer} nodeContainer`}>
          <HirearchyNode client={node} isSpread={isSpread} showChilds={showChildrenNode} />
          <div className={classes.subLevelContainer} style={{ display: isSpread ? 'block' : 'none' }}>
            {node.sponsoredCompaniesIds
              .sort((a, b) => (hirearchyMap[b].createdAt || '').localeCompare(hirearchyMap[a].createdAt || ''))
              .map(id => drawNode(hirearchyMap[id]))}
          </div>
        </div>
      );
    },
    [childrenList, classes, hirearchyMap, showChildrenNode]
  );

  useEffect(() => {
    /* istanbul ignore else */
    if (assignClientLoading && !assignClientLoading.isLoading) {
      closeModal();
      fetchProjectClientHirearchyList(projectId);
    }
  }, [assignClientLoading, projectId, closeModal, fetchProjectClientHirearchyList]);

  useEffect(() => {
    fetchProjectClientHirearchyList(projectId);
  }, [projectId, fetchProjectClientHirearchyList]);

  if (!!loading?.isLoading) return <>Loading...</>;
  return <div className={classes.container}>{hirearchyList.map(client => drawNode(client))}</div>;
};

export default memo(HirearchyView);
