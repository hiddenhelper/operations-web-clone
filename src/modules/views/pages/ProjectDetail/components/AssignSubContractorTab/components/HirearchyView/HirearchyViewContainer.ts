import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import { GENERAL } from '../../../../../../../../constants';
import { IRootState } from '../../../../../../../state-mgmt/rootState';
import { clientState } from '../../../../../../../state-mgmt/client';

import HirearchyView from './HirearchyView';

export const mapStateToProps = (state: IRootState) => ({
  hirearchyMap: state.client.clientProjectHirearchyMap,
  loading: state.general.loadingMap[GENERAL.LOADING_KEY.FETCH_PROJECT_CLIENT_HIREARCHY_LIST],
  assignClientLoading: state.general.loadingMap[GENERAL.LOADING_KEY.ASSIGN_CLIENT_PROJECT],
});

export const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchProjectClientHirearchyList: (id: string) => dispatch(clientState.actions.fetchProjectClientHirearchyListStart(id)),
  clearClientMap: () => dispatch(clientState.actions.clearClientMap()),
});

export default connect(mapStateToProps, mapDispatchToProps)(HirearchyView);
