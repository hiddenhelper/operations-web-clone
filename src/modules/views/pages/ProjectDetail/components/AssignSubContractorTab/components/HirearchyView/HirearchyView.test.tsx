import React from 'react';
import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';
import { MemoryRouter } from 'react-router';
import { render, fireEvent, act } from '@testing-library/react';

import { UserModel } from '../../../../../../../models';
import {
  getProject_1,
  getClientProjectHirearchy_1,
  getClientProjectHirearchy_2,
  getClientProjectHirearchy_3,
  getClientProjectHirearchy_4,
} from '../../../../../../../../test/entities';
import { getInitialState, getAdminInitialState } from '../../../../../../../../test/rootState';
import HirearchyView, { IHirearchyViewProps } from './HirearchyView';

describe('HirearchyView', () => {
  let props: IHirearchyViewProps;

  beforeEach(() => {
    props = {
      projectId: getProject_1().id,
      hirearchyMap: {
        [getClientProjectHirearchy_1().id]: getClientProjectHirearchy_1(),
        [getClientProjectHirearchy_2().id]: getClientProjectHirearchy_2(),
        [getClientProjectHirearchy_3().id]: getClientProjectHirearchy_3(),
        [getClientProjectHirearchy_4().id]: getClientProjectHirearchy_4(),
      },
      loading: {
        isLoading: false,
        hasError: false,
        error: null,
      },
      assignClientLoading: undefined,
      fetchProjectClientHirearchyList: jest.fn(),
      closeModal: jest.fn(),
    };
  });

  it('should fetchProjectClientHirearchyList', () => {
    render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <HirearchyView {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(props.fetchProjectClientHirearchyList).toHaveBeenCalledWith(props.projectId);
  });

  it('should render', () => {
    const { container } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <HirearchyView {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render loading', () => {
    props.loading = { isLoading: true, hasError: false, error: null };
    const { container } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <HirearchyView {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should close modal', () => {
    props.assignClientLoading = { isLoading: false, hasError: false, error: null };
    render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <HirearchyView {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(props.closeModal).toHaveBeenCalled();
    expect(props.fetchProjectClientHirearchyList).toHaveBeenCalled();
  });

  it('should render children nodes', () => {
    const { container, getAllByTestId } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <HirearchyView {...props} />
        </MemoryRouter>
      </Provider>
    );

    act(() => {
      fireEvent.click(getAllByTestId('open-hirearchy-child-list')[0]);
    });

    expect(container).toMatchSnapshot();
  });
});
