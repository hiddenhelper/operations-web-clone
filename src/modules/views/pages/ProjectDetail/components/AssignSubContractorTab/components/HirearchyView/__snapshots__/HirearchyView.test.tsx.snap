// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HirearchyView should render 1`] = `
<div>
  <div
    class="makeStyles-container-44"
  >
    <div
      class="makeStyles-nodeContainer-45 nodeContainer"
    >
      <div
        class="makeStyles-node-47"
      >
        <div
          class="makeStyles-nodeContent-48"
        >
          <div
            class="spread-icon"
            style="min-width: 30px;"
          >
            <button
              aria-label="hirearchy-client-node"
              class="MuiButtonBase-root MuiIconButton-root"
              data-testid="open-hirearchy-child-list"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-role-51"
          >
            <svg
              height="70"
              width="70"
            >
              <circle
                cx="36"
                cy="36"
                fill="#E5E5E5"
                r="18"
              />
              <text
                fill="#AAAAAA"
                font-family="Verdana"
                font-size="13"
                font-weight="bold"
                x="26"
                y="41"
              >
                GC
              </text>
            </svg>
          </div>
          <div
            class="makeStyles-listName-70 makeStyles-listGeneralText-72"
          >
            Martin Fowler
          </div>
          <div
            class="role"
          >
            General Contractor
          </div>
          <div
            class="makeStyles-nodeTrades-50"
          >
            trade 1, trade 2, trade 3
          </div>
        </div>
        <div
          class="makeStyles-nodeStatus-49"
          style="display: flex;"
        >
          <div
            class="MuiChip-root makeStyles-statusChip-52 makeStyles-active-58  MuiChip-colorPrimary"
          >
            <span
              class="MuiChip-label"
            >
              Accepted
            </span>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-subLevelContainer-46"
        style="display: none;"
      >
        <div
          class="makeStyles-nodeContainer-45 nodeContainer"
        >
          <div
            class="makeStyles-node-47"
          >
            <div
              class="makeStyles-nodeContent-48"
            >
              <div
                class="spread-icon"
                style="min-width: 30px;"
              >
                <button
                  aria-label="hirearchy-client-node"
                  class="MuiButtonBase-root MuiIconButton-root"
                  data-testid="open-hirearchy-child-list"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <div
                class="makeStyles-role-51"
              >
                <svg
                  height="70"
                  width="70"
                >
                  <circle
                    cx="36"
                    cy="36"
                    fill="#E5E5E5"
                    r="18"
                  />
                  <text
                    fill="#AAAAAA"
                    font-family="Verdana"
                    font-size="11"
                    font-weight="bold"
                    x="24"
                    y="40"
                  >
                    DEV
                  </text>
                </svg>
              </div>
              <div
                class="makeStyles-listName-70 makeStyles-listGeneralText-72"
              >
                
              </div>
              <div
                class="role"
              >
                Developer
              </div>
              <div
                class="makeStyles-nodeTrades-50"
              />
            </div>
            <div
              class="makeStyles-nodeStatus-49"
              style="display: flex;"
            >
              <div
                class="MuiChip-root makeStyles-statusChip-52 makeStyles-active-58  MuiChip-colorPrimary"
              >
                <span
                  class="MuiChip-label"
                >
                  Accepted
                </span>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-subLevelContainer-46"
            style="display: none;"
          >
            <div
              class="makeStyles-nodeContainer-45 nodeContainer"
            >
              <div
                class="makeStyles-node-47"
              >
                <div
                  class="makeStyles-nodeContent-48"
                >
                  <div
                    class="spread-icon"
                    style="min-width: 30px;"
                  />
                  <div
                    class="makeStyles-role-51"
                  >
                    <svg
                      height="70"
                      width="70"
                    >
                      <circle
                        cx="36"
                        cy="36"
                        fill="#E5E5E5"
                        r="18"
                      />
                      <text
                        fill="#AAAAAA"
                        font-family="Verdana"
                        font-size="11"
                        font-weight="bold"
                        x="24"
                        y="40"
                      >
                        DEV
                      </text>
                    </svg>
                  </div>
                  <div
                    class="makeStyles-listName-70 makeStyles-listGeneralText-72"
                  >
                    Robert C. Martin
                  </div>
                  <div
                    class="role"
                  >
                    Developer
                  </div>
                  <div
                    class="makeStyles-nodeTrades-50"
                  >
                    trade 1, trade 2
                  </div>
                </div>
                <div
                  class="makeStyles-nodeStatus-49"
                  style="display: flex;"
                >
                  <div
                    class="MuiChip-root makeStyles-statusChip-52 makeStyles-pending-59  MuiChip-colorPrimary"
                  >
                    <span
                      class="MuiChip-label"
                    >
                      Pending
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-subLevelContainer-46"
                style="display: none;"
              />
            </div>
            <div
              class="makeStyles-nodeContainer-45 nodeContainer"
            >
              <div
                class="makeStyles-node-47"
              >
                <div
                  class="makeStyles-nodeContent-48"
                >
                  <div
                    class="spread-icon"
                    style="min-width: 30px;"
                  />
                  <div
                    class="makeStyles-role-51"
                  >
                    <svg
                      height="70"
                      width="70"
                    >
                      <circle
                        cx="36"
                        cy="36"
                        fill="#E5E5E5"
                        r="18"
                      />
                      <text
                        fill="#AAAAAA"
                        font-family="Verdana"
                        font-size="11"
                        font-weight="bold"
                        x="24"
                        y="40"
                      >
                        DEV
                      </text>
                    </svg>
                  </div>
                  <div
                    class="makeStyles-listName-70 makeStyles-listGeneralText-72"
                  >
                    Robert C. Martin
                  </div>
                  <div
                    class="role"
                  >
                    Developer
                  </div>
                  <div
                    class="makeStyles-nodeTrades-50"
                  >
                    trade 1, trade 2
                  </div>
                </div>
                <div
                  class="makeStyles-nodeStatus-49"
                  style="display: flex;"
                >
                  <div
                    class="MuiChip-root makeStyles-statusChip-52 makeStyles-pending-59  MuiChip-colorPrimary"
                  >
                    <span
                      class="MuiChip-label"
                    >
                      Pending
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-subLevelContainer-46"
                style="display: none;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HirearchyView should render children nodes 1`] = `
<div>
  <div
    class="makeStyles-container-138"
  >
    <div
      class="makeStyles-nodeContainer-139 nodeContainer"
    >
      <div
        class="makeStyles-node-141"
      >
        <div
          class="makeStyles-nodeContent-142"
        >
          <div
            class="spread-icon"
            style="min-width: 30px;"
          >
            <button
              aria-label="hirearchy-client-node"
              class="MuiButtonBase-root MuiIconButton-root"
              data-testid="open-hirearchy-child-list"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-role-145"
          >
            <svg
              height="70"
              width="70"
            >
              <circle
                cx="36"
                cy="36"
                fill="#E5E5E5"
                r="18"
              />
              <text
                fill="#AAAAAA"
                font-family="Verdana"
                font-size="13"
                font-weight="bold"
                x="26"
                y="41"
              >
                GC
              </text>
            </svg>
          </div>
          <div
            class="makeStyles-listName-164 makeStyles-listGeneralText-166"
          >
            Martin Fowler
          </div>
          <div
            class="role"
          >
            General Contractor
          </div>
          <div
            class="makeStyles-nodeTrades-144"
          >
            trade 1, trade 2, trade 3
          </div>
        </div>
        <div
          class="makeStyles-nodeStatus-143"
          style="display: flex;"
        >
          <div
            class="MuiChip-root makeStyles-statusChip-146 makeStyles-active-152  MuiChip-colorPrimary"
          >
            <span
              class="MuiChip-label"
            >
              Accepted
            </span>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-subLevelContainer-140"
        style="display: block;"
      >
        <div
          class="makeStyles-nodeContainer-139 nodeContainer"
        >
          <div
            class="makeStyles-node-141"
          >
            <div
              class="makeStyles-nodeContent-142"
            >
              <div
                class="spread-icon"
                style="min-width: 30px;"
              >
                <button
                  aria-label="hirearchy-client-node"
                  class="MuiButtonBase-root MuiIconButton-root"
                  data-testid="open-hirearchy-child-list"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <div
                class="makeStyles-role-145"
              >
                <svg
                  height="70"
                  width="70"
                >
                  <circle
                    cx="36"
                    cy="36"
                    fill="#E5E5E5"
                    r="18"
                  />
                  <text
                    fill="#AAAAAA"
                    font-family="Verdana"
                    font-size="11"
                    font-weight="bold"
                    x="24"
                    y="40"
                  >
                    DEV
                  </text>
                </svg>
              </div>
              <div
                class="makeStyles-listName-164 makeStyles-listGeneralText-166"
              >
                
              </div>
              <div
                class="role"
              >
                Developer
              </div>
              <div
                class="makeStyles-nodeTrades-144"
              />
            </div>
            <div
              class="makeStyles-nodeStatus-143"
              style="display: flex;"
            >
              <div
                class="MuiChip-root makeStyles-statusChip-146 makeStyles-active-152  MuiChip-colorPrimary"
              >
                <span
                  class="MuiChip-label"
                >
                  Accepted
                </span>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-subLevelContainer-140"
            style="display: none;"
          >
            <div
              class="makeStyles-nodeContainer-139 nodeContainer"
            >
              <div
                class="makeStyles-node-141"
              >
                <div
                  class="makeStyles-nodeContent-142"
                >
                  <div
                    class="spread-icon"
                    style="min-width: 30px;"
                  />
                  <div
                    class="makeStyles-role-145"
                  >
                    <svg
                      height="70"
                      width="70"
                    >
                      <circle
                        cx="36"
                        cy="36"
                        fill="#E5E5E5"
                        r="18"
                      />
                      <text
                        fill="#AAAAAA"
                        font-family="Verdana"
                        font-size="11"
                        font-weight="bold"
                        x="24"
                        y="40"
                      >
                        DEV
                      </text>
                    </svg>
                  </div>
                  <div
                    class="makeStyles-listName-164 makeStyles-listGeneralText-166"
                  >
                    Robert C. Martin
                  </div>
                  <div
                    class="role"
                  >
                    Developer
                  </div>
                  <div
                    class="makeStyles-nodeTrades-144"
                  >
                    trade 1, trade 2
                  </div>
                </div>
                <div
                  class="makeStyles-nodeStatus-143"
                  style="display: flex;"
                >
                  <div
                    class="MuiChip-root makeStyles-statusChip-146 makeStyles-pending-153  MuiChip-colorPrimary"
                  >
                    <span
                      class="MuiChip-label"
                    >
                      Pending
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-subLevelContainer-140"
                style="display: none;"
              />
            </div>
            <div
              class="makeStyles-nodeContainer-139 nodeContainer"
            >
              <div
                class="makeStyles-node-141"
              >
                <div
                  class="makeStyles-nodeContent-142"
                >
                  <div
                    class="spread-icon"
                    style="min-width: 30px;"
                  />
                  <div
                    class="makeStyles-role-145"
                  >
                    <svg
                      height="70"
                      width="70"
                    >
                      <circle
                        cx="36"
                        cy="36"
                        fill="#E5E5E5"
                        r="18"
                      />
                      <text
                        fill="#AAAAAA"
                        font-family="Verdana"
                        font-size="11"
                        font-weight="bold"
                        x="24"
                        y="40"
                      >
                        DEV
                      </text>
                    </svg>
                  </div>
                  <div
                    class="makeStyles-listName-164 makeStyles-listGeneralText-166"
                  >
                    Robert C. Martin
                  </div>
                  <div
                    class="role"
                  >
                    Developer
                  </div>
                  <div
                    class="makeStyles-nodeTrades-144"
                  >
                    trade 1, trade 2
                  </div>
                </div>
                <div
                  class="makeStyles-nodeStatus-143"
                  style="display: flex;"
                >
                  <div
                    class="MuiChip-root makeStyles-statusChip-146 makeStyles-pending-153  MuiChip-colorPrimary"
                  >
                    <span
                      class="MuiChip-label"
                    >
                      Pending
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-subLevelContainer-140"
                style="display: none;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HirearchyView should render loading 1`] = `
<div>
  Loading...
</div>
`;
